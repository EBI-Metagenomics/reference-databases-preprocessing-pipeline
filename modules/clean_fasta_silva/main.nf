
process CLEAN_FASTA_SILVA {

    label 'light'
    container = '/hps/nobackup/rdf/metagenomics/service-team/singularity-cache/quay.io_biocontainers_seqtk:1.3.sif'
    publishDir "${params.outdir}/${label}/${version}/", mode: 'copy'

    input:
    path fasta
    path tax
    val version
    val label

    output:
    path("${label}.fasta"), emit: cleaned_fasta

    """
    grep -v '^#' $tax | cut -f1 > ${label}.idlst
    seqtk subseq $fasta ${label}.idlst > ${label}.fasta
    """

}