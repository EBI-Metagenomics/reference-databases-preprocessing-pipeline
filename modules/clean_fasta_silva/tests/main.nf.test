nextflow_process {

    name "Test Process CLEAN_FASTA_SILVA"
    script "modules/clean_fasta_silva/main.nf"
    process "CLEAN_FASTA_SILVA"
    tag "modules"
    tag "clean_fasta_silva"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = file('${projectDir}/modules/clean_fasta_silva/tests/data/test.fasta', checkIfExists: true)
                input[1] = file('${projectDir}/modules/clean_fasta_silva/tests/data/test-tax.txt', checkIfExists: true)
                input[2] = params.silva_version
                input[3] = params.silva_ssu_label
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.cleaned_fasta).match("cleaned_fasta") }
            )
        }

    }

}
