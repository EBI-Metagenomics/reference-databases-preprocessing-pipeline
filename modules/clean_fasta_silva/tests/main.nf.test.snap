{
    "cleaned_fasta": {
        "content": [
            [
                "SILVA-SSU.fasta:md5,68c60c9c9b78c0244f05e78fac9926a6"
            ]
        ],
        "timestamp": "2023-12-12T10:09:21.174627208"
    }
}